plugin/src/test/java/org/opensearch/ml/rest/MLCommonsRestTestCase.java

```diff
@@ -464,4 +464,67 @@ public void trainAndPredict(
             function.accept(predictionResult);
         }
     }
+
+    public void train(
+        RestClient client,
+        FunctionName functionName,
+        String indexName,
+        MLAlgoParams params,
+        SearchSourceBuilder searchSourceBuilder,
+        Consumer<Map<String, Object>> function,
+        boolean async
+    ) throws IOException {
+        MLInputDataset inputData = SearchQueryInputDataset
+            .builder()
+            .indices(ImmutableList.of(indexName))
+            .searchSourceBuilder(searchSourceBuilder)
+            .build();
+        MLInput kmeansInput = MLInput.builder().algorithm(functionName).parameters(params).inputDataset(inputData).build();
+        String endpoint = "/_plugins/_ml/_train/" + functionName.name().toLowerCase(Locale.ROOT);
```

Comment:
nit: Should this be a constant, same for predict endpoint

Reply:
Prefer to keep it as is now as we are quite close to release. We can refactor code later. Created an issue to track https://github.com/opensearch-project/ml-commons/issues/178